File: Data Structures & Algorithms/remove-element/submission-4.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        k = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[k] = nums[i]
                nums[i] = nums[k]
                k += 1

        print(nums)
        return k


solution = Solution()
print(solution.removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2))

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        HashSet<Integer> set = new HashSet<>();
        for (int i : nums) {
            set.add(i);
        }
        int res = 1;
        for (int i : nums) {

            int cur = 1;
            int j = i;
            if (!set.contains(j - 1)) {
                while (set.contains(j + 1)) {
                    cur++;
                    j++;
                }
                if (cur > res) {
                    res = cur;
                }
            }
        }
        return res;
    }
}
